### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by AlexBondor**

### CODE REVIEW

File: src/main/java/com/places/server/repository/impl/AbstractRepositoryImpl.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  @Inject
+  private Provider<EntityManager> entityManagerProvider;
+
+  @Override
+  public T save(T entity) {
+    return entityManagerProvider.get().merge(entity);
+  }
+
+  @Override
+  public T findById(Long id) {
```
Comment (radumarias) [inline src/main/java/com/places/server/repository/impl/AbstractRepositoryImpl.java]: you can directly use `javax.persistence.EntityManager#find(java.lang.Class<T>, java.lang.Object)`

File: src/main/java/com/places/server/config/AppPersistenceInitializer.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+import com.google.inject.persist.PersistService;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+@Singleton
+class AppPersistenceInitializer {
+
+  @Inject
+  AppPersistenceInitializer(final PersistService persistService) {
+    persistService.start();
```
[the remark this answers is not in the document]
Comment (AlexBondor) [reply src/main/java/com/places/server/config/AppPersistenceInitializer.java]: Yes.. but then I'd need to do end to end tests as the entityManager won't get initialized for the integration test I've written

File: src/main/java/com/places/server/repository/impl/AbstractRepositoryImpl.java (hunk 1 above)
Comment (AlexBondor) [reply src/main/java/com/places/server/repository/impl/AbstractRepositoryImpl.java]: 👍

File: src/main/java/com/places/server/config/AppPersistenceInitializer.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (radumarias) [reply src/main/java/com/places/server/config/AppPersistenceInitializer.java]: for tests you will have different Test module where you can initialize, but for prod it's best to use theirs

**Comment by AlexBondor** [comment]
I've added Google Maps Places API client and I've created the dashboard component with some basic functionality (searching for cities using Places API)

File: src/main/java/com/places/server/config/AppModule.java [hunk 3]
```diff
@@ -1,22 +1,34 @@
...
 import com.google.inject.persist.jpa.JpaPersistModule;
+import com.places.client.service.PlaceService;
 import com.places.server.repository.PlaceRepository;
 import com.places.server.repository.impl.PlaceRepositoryImpl;
-import com.places.server.service.PlaceService;
 import com.places.server.service.impl.PlaceServiceImpl;
+import java.io.IOException;
+import java.util.Properties;
 
-public class AppModule extends PrivateModule {
```
Comment (radumarias) [inline src/main/java/com/places/server/config/AppModule.java]: you can extend Module then you don't need to expose explicitly

**Comment by AlexBondor** [comment]
Should we get this merged before I push some new changes? It would be better that way I would say.. otherwise we'll get loads of changes..

**Pull request merged**